File: services/customers-service/src/test/java/br/com/deltaglobalbank/customers/domain/customer/valueobjects/MotherNameTests.java
package br.com.deltaglobalbank.customers.domain.customer.valueobjects;

import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertThrows;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.ValueSource;

class MotherNameTests {

    @Test
    void mustAcceptAValidMotherName() {
        assertDoesNotThrow(() -> new MotherName("Ana Silva"));
    }

    @ParameterizedTest
    @ValueSource(strings = {"", " "})
    void mustRejectBlankMotherName(String blank) {
        assertThrows(IllegalArgumentException.class, () -> new MotherName(blank));
    }

    @Test
    void mustRejectMotherNameExceedingMaxLength() {
        assertThrows(IllegalArgumentException.class, () -> new MotherName("a".repeat(256)));
    }

    @Test
    void mustTrimSurroundingSpaces() {
        assertEquals("Ana Silva", new MotherName("  Ana Silva  ").value());
    }
}

File: services/customers-service/src/test/kotlin/br/com/deltaglobalbank/customers/domain/customer/valueobjects/MotherNameTests.kt
package br.com.deltaglobalbank.customers.domain.customer.valueobjects

import org.junit.jupiter.api.Assertions.assertDoesNotThrow
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertThrows
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.ValueSource
import org.junit.jupiter.api.Test

class MotherNameTests {

    @Test
    fun `must accept a valid mother name`() {
        assertDoesNotThrow { MotherName("Ana Silva") }
    }

    @ParameterizedTest
    @ValueSource(strings = ["", " "])
    fun `must reject blank mother name`(blank: String) {
        assertThrows(IllegalArgumentException::class.java) { MotherName(blank) }
    }

    @Test
    fun `must reject mother name exceeding max length`() {
        assertThrows(IllegalArgumentException::class.java) { MotherName("a".repeat(256)) }
    }

    @Test
    fun `must trim surrounding spaces`() {
        assertEquals("Ana Silva", MotherName("  Ana Silva  ").value)
    }
}